pkg/kubelet/util/util_windows_test.go

```diff
@@ -90,3 +96,103 @@ func TestParseEndpoint(t *testing.T) {
 	}
 
 }
+
+func TestIsUnixDomainSocket(t *testing.T) {
+	tests := []struct {
+		label          string
+		listenOnSocket bool
+		regularFile    bool
+		expectSocket   bool
+		expectError    bool
+		invalidFile    bool
+		listenOnPipe   bool
+	}{
+		{
+			label:          "Domain Socket File",
+			listenOnSocket: true,
+			expectSocket:   true,
+			expectError:    false,
+		},
+		{
+			label:        "Non Existent File",
+			invalidFile:  true,
+			expectSocket: false,
+			expectError:  true,
+		},
+		{
+			label:        "Regular File",
+			regularFile:  true,
+			expectSocket: false,
+			expectError:  false,
+		},
+		{
+			label:        "Named Pipe",
+			listenOnPipe: true,
+			expectError:  true,
+			expectSocket: false,
+		},
+	}
+	for _, test := range tests {
+		var f *os.File
+		var addr string
+		var unixln *net.UnixListener
+		var pipeln net.Listener
+		var err error
+		if test.listenOnPipe {
+			// generate a random pipe name to listen on
```

Comment:
nit: define a helper function `getRandomPipeName := func(prefix string, len int) string{...` may make the test code more readable

Reply:
Done.